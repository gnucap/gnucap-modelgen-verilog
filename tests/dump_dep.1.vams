`include "disciplines.vams"
`include "constants.vams"
`pragma modelgen dump-annotate
`pragma modelgen nodump-nature
`pragma modelgen nodump-discipline

module dep0(d, g);
  inout      d, g;
  electrical d, g;
  analog begin : main
    real c;
    real T;
    real S;
    S = T;
    while(1) begin
      c = T;
      T = V(g);
    end
    S = T;
  end

endmodule
