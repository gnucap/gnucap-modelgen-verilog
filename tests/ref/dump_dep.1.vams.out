module dep0(d, g);
  inout d, g;
  electrical d, g;
  analog begin : main // always
    // S : 1
    // T : 0
    // c : 0
    real c;
    real T;
    real S;
    S = T; // s3
    while (1) begin
      // T : 1
      // c : 1
      c = T; // dep: _potential_b_g_ s3
      T = V(g); // dep: _potential_b_g_ s1
    end
    S = T; // dep: _potential_b_g_ s3
  end

endmodule



